heron/tools/cli/src/python/result.py

```diff
@@ -0,0 +1,188 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''result.py'''
+import abc
+import sys
+from enum import Enum
+
+from heron.common.src.python.utils.log import Log
+
+# Meaning of exit status code:
+#  - status code = 0:
+#    program exits without error
+#  - 0 < status code < 100:
+#    program fails to execute before program execution. For example,
+#    JVM cannot find or load main class
+#  - 100 <= status code < 200:
+#    program fails to launch after program execution. For example,
+#    topology definition file fails to be loaded
+#  - status code == 200:
+#    program sends out dry-run response
+
+# Definition corresponds to definition in com.twitter.heron.scheduler.AbstractMain
+
+# pylint: disable=no-init
+class Status(Enum):
+  """Status code enum"""
+  Ok = 0
+  InvocationError = 1
+  HeronError = 100
+  DryRun = 200
+
+def status_type(status_code):
+  if status_code == 0:
+    return Status.Ok
+  elif status_code < 100:
+    return Status.InvocationError
+  elif status_code == 200:
+    return Status.DryRun
+  else:
+    return Status.HeronError
+
+class Result(object):
+  """Result class"""
+  def __init__(self, status=None, err_context=None, succ_context=None):
+    self.status = status
+    self.err_context = err_context
+    self.succ_context = succ_context
+
+  @staticmethod
+  def _do_log(log_f, msg):
+    if msg:
+      if msg[-1] == '\n':
+        msg = msg[:-1]
+      log_f(msg)
+
+  def _log_context(self):
+    # render context only after process exits
+    assert self.status is not None
+    if self.status == Status.Ok or self.status == Status.DryRun:
+      self._do_log(Log.info, self.succ_context)
+    elif self.status == Status.HeronError:
+      self._do_log(Log.error, self.err_context)
+    elif self.status == Status.InvocationError:
+      # invocation error has no context
+      pass
+    else:
+      raise RuntimeError(
+          "Unknown status type of value %d. Expected value: %s", self.status.value, list(Status))
+
+  def add_context(self, err_context, succ_context=None):
+    """ Prepend msg to add some context information
+
+    :param pmsg: context info
+    :return: None
+    """
+    self.err_context = err_context
+    self.succ_context = succ_context
+
+  def is_successful(self):
+    return self.status == Status.Ok
+
+  @abc.abstractmethod
+  def render(self):
+    pass
+
+
+class SimpleResult(Result):
+  """Simple result: result that already and only
+     contains status of the result"""
+  def __init__(self, status):
+    super(SimpleResult, self).__init__(status)
+
+  def render(self):
+    self._log_context()
+
+
+class ProcessResult(Result):
+  """Process result: a wrapper of result class"""
+  def __init__(self, proc):
+    super(ProcessResult, self).__init__()
+    self.proc = proc
+
+  def renderProcessStdErr(self, stderr_line):
```

Comment:
This logic seems sound if it works in your tests.

In the future, best to avoid making code changes _and_ doing a file rename at the same time, since there's no longer a diff to review.